src/Servers/HttpSys/test/FunctionalTests/OpaqueUpgradeTests.cs

```diff
@@ -267,7 +268,36 @@ public async Task OpaqueUpgrade_InvalidMethodUpgrade_Disconnected(string method,
                 }
             }))
             {
-                await Assert.ThrowsAsync<InvalidOperationException>(async () => await SendOpaqueRequestAsync(method, address, extraHeader));
+                var ex = await Assert.ThrowsAsync<InvalidOperationException>(async () => await SendOpaqueRequestAsync(method, address, extraHeader));
+                Assert.Equal("The response status code was incorrect: HTTP/1.1 200 OK", ex.Message);
+            }
+        }
+
+        [ConditionalFact]
+        [MinimumOSVersion(OperatingSystems.Windows, WindowsVersions.Win8)]
+        public async Task OpaqueUpgrade_PostWithBodyAndUpgradeHeaders_Accepted()
+        {
+            using (Utilities.CreateHttpServer(out string address, async httpContext =>
+            {
+                try
+                {
+                    var opaqueFeature = httpContext.Features.Get<IHttpUpgradeFeature>();
+                    Assert.NotNull(opaqueFeature);
+                    Assert.False(opaqueFeature.IsUpgradableRequest);
+                }
+                catch (Exception ex)
+                {
+                    await httpContext.Response.WriteAsync(ex.ToString());
+                }
+            }))
+            {
+                using var client = new HttpClient();
+
+                var request = new HttpRequestMessage(HttpMethod.Post, address);
+                request.Headers.Connection.Add("upgrade");
```

Comment:
Should we verify this works without the `Connection: Upgrade` header?

Reply:
? This is testing that Http.Sys doesn't reject requests that have a `Connection: Upgrade` and a body, like Kestrel used to. Without the Connection header it's a regular Post request.

The all of the actual upgrade tests call SendOpaqueRequestAsync which does not include the Connection header and still works.